code/modules/vehicles/mecha/equipment/tools/medical_tools.dm

```diff
@@ -32,34 +32,68 @@
 		content.forceMove(get_turf(src))
 	return ..()
 
+/obj/item/mecha_parts/mecha_equipment/medical/proc/get_reagent_data(list/datum/reagent/reagent_list)
+	var/list/contained_reagents = list()
+	if(length(reagent_list))
+		for(var/datum/reagent/reagent as anything in reagent_list)
+			contained_reagents += list(list("name" = reagent.name, "volume" = round(reagent.volume, 0.01))) // list in a list because Byond merges the first list...
+	return contained_reagents
+
 /obj/item/mecha_parts/mecha_equipment/medical/sleeper/get_snowflake_data()
 	var/list/data = list("snowflake_id" = MECHA_SNOWFLAKE_ID_SLEEPER)
-	if(!patient)
+	if(isnull(patient))
 		return data
+	var/patient_state
+	switch(patient.stat)
+		if(0)
+			patient_state = "Conscious"
+		if(1)
+			patient_state = "Unconscious"
+		if(2)
+			patient_state = "*dead*"
+		else
+			patient_state = "Unknown"
+	var/core_temp = ""
+	if(ishuman(patient))
+		var/mob/living/carbon/human/humi = patient
+		core_temp = humi.bodytemperature-T0C
 	data["patient"] = list(
-		"patientname" = patient.name,
-		"is_dead" = patient.stat == DEAD,
+		"patient_name" = patient.name,
 		"patient_health" = patient.health/patient.maxHealth,
+		"patient_state" = patient_state,
+		"core_temp" = core_temp,
+		"brute_loss" = patient.getBruteLoss(),
+		"burn_loss" = patient.getFireLoss(),
+		"toxin_loss" = patient.getToxLoss(),
+		"oxygen_loss" = patient.getOxyLoss(),
 	)
+	data["has_brain_damage"] = patient.get_organ_loss(ORGAN_SLOT_BRAIN) != 0
+	data["has_traumas"] = length(patient.get_traumas()) != 0
+	data["contained_reagents"] = get_reagent_data(patient.reagents.reagent_list)
+
+	var/obj/item/mecha_parts/mecha_equipment/medical/syringe_gun/SG = locate(/obj/item/mecha_parts/mecha_equipment/medical/syringe_gun) in chassis
+	data["injectible_reagents"] = get_reagent_data(SG.reagents.reagent_list)
 	return data
 
 /obj/item/mecha_parts/mecha_equipment/medical/sleeper/handle_ui_act(action, list/params)
 	switch(action)
 		if("eject")
 			go_out()
 			return TRUE
-		if("view_stats")
-			usr << browse(get_patient_stats(),"window=msleeper")
-			onclose(usr, "msleeper")
-			return FALSE
+		else
+			var/obj/item/mecha_parts/mecha_equipment/medical/syringe_gun/SG = locate() in chassis
+			for(var/datum/reagent/R in SG.reagents.reagent_list)
+				if(action == ("inject_reagent_" + R.name))
+					inject_reagent(R, SG)
+					return
```

Comment:
we can clean up the indentation here since a lot of this is un-necessary to be in an `else` case.

```suggestion
	var/obj/item/mecha_parts/mecha_equipment/medical/syringe_gun/shooter = locate() in chassis
	for(var/datum/reagent/medication in shooter.reagents.reagent_list)
		if(action == ("inject_reagent_" + medication.name))
			inject_reagent(medication, shooter)
			break // or maybe return TRUE? i'm not certain
				
	return FALSE
```